app/javascript/packages/transliterable-field-group/index.ts

```diff
@@ -0,0 +1,375 @@
+import { trackError } from '@18f/identity-analytics';
+import { request } from '@18f/identity-request';
+
+/**
+ * Representation of the response from a transliteration API call
+ */
+interface ValidationResponse {
+  success: boolean;
+  data: Record<string, string>;
+}
+
+/**
+ * Internal cache structure for controlling when API call is made
+ * and reusing result
+ */
+interface ValidationResultCache {
+  preValidationValues: Record<string, string>;
+  postValidationValues: Record<string, string>;
+  result: Record<string, string>;
+}
+
+/**
+ * This adds an asynchronous transliteration check for certain form fields
+ * that gets run between form validation and submission.
+ */
+class TransliterableFieldGroupElement extends HTMLElement {
+  private static readonly DEFAULT_INPUT_TIMEOUT_MS = 1500;
+
+  /**
+   * Cache to moderate API calls and allow reuse of the previous result
```

Comment:
moderate in this description - is it the noun or the verb?

Reply:
This is a verb.